### ISSUE TITLE
datsets typo causing push issue

### ISSUE THREAD

**Issue by blackdotbug** [see-mike-out/erie-web#5]
I'm trying to compile an audiograph for an overlay and I'm getting this issue: 

```
TypeError: datasets.push is not a function
    at normalizeSpecification (audio-graph-normalize.js:34:16)
    at Module.compileAudioGraph (audio-graph.js:18:106)
```

I think it might be because datasets is misspelled `datsets` in a few places:  
https://github.com/see-mike-out/erie-web/blob/8255b2bb609ce135d71c0211fe111e7416af5ffa/src/classes/erie-datasets.js#L6

I'll try to submit a PR.

**Comment by blackdotbug** [comment]
I don't have permissions to push a branch to start a pull request, but I went a little deeper into this problem and discovered a few things I had to change to get it working in my application. 

In this section I added some more conditions that prevent a null dataset from getting pushed into datasets. 
https://github.com/see-mike-out/erie-web/blob/8255b2bb609ce135d71c0211fe111e7416af5ffa/src/compile/audio-graph-normalize.js#L32

```javascript
    if (spec.data && !spec.data.name && spec.data.type !== "unset" && spec.data.values) {
```

you probably don't need all four of those checks, but I wanted to be real sure it wasn't putting something in that shouldn't be there. I added the same set of conditions here: 
https://github.com/see-mike-out/erie-web/blob/8255b2bb609ce135d71c0211fe111e7416af5ffa/src/compile/audio-graph-normalize.js#L43

then I added another condition to the getData function so the data values were grabbed if they were embedded in a data object with a name. 
https://github.com/see-mike-out/erie-web/blob/8255b2bb609ce135d71c0211fe111e7416af5ffa/src/data/audio-graph-data-import.js#L21

```javascript
  if (data_spec?.values) {
    return data_spec.values;
  } else if (data_spec?.data?.values) {
    return data_spec.data.values;
  } else if (data_spec?.csv) {
```

And that got the overlay working in my use case. Hope this helps!
